plonky3/src/circuit_builder.rs

```diff
@@ -28,17 +30,24 @@ impl<'a, T: FieldElement> PowdrCircuit<'a, T> {
     pub fn generate_trace_rows(&self) -> RowMajorMatrix<Goldilocks> {
         // an iterator over all columns, committed then fixed
         let witness = self.witness().iter();
+        let publics = self.publics.iter();
         let len = self.analyzed.degree.unwrap();
 
         // for each row, get the value of each column
         let values = (0..len)
             .flat_map(move |i| {
+                // witness values
                 witness
+                .clone()
+                .map(move |(_, v)| cast_to_goldilocks(v[i as usize]))
+                .chain(
+                    publics
                     .clone()
```

Comment:
I think this clone is not necessary. Are you running clippy? There is a vscode extension which would catch things like this.

Reply:
Also I'd be surprised if all these `move` were required

Reply:
Ah I was not, I just copied the syntax from the witness statement above, I'll also remove the `move` in publics since it seems to just be matching integers